File: src/lessons/welcome/bat/bool2/CaughtSpeedingEntity.java
package lessons.welcome.bat.bool2;

import static plm.core.ValueSerializer.deserialize;
import static plm.core.ValueSerializer.serialize;

import plm.universe.bat.BatEntity;

public class CaughtSpeedingEntity extends BatEntity {

  public void run()
  {
    int count = getTestCount();
    for (int i = 0; i < count; i++) {
      Object[] param = (Object[])deserialize(getTest(i));
      setTestResult(i, serialize(caughtSpeeding((Integer)param[0], (Boolean)param[1])));
    }
  }

  /* BEGIN TEMPLATE */
  int caughtSpeeding(int speed, boolean isBirthday)
  {
    /* BEGIN SOLUTION */
    if ((isBirthday && speed <= 65) || (speed <= 60))
      return 0;
    else if ((isBirthday && speed <= 85) || (speed <= 80))
      return 1;
    else
      return 2;
    /* END SOLUTION */
  }
  /* END TEMPLATE */
}

File: src/lessons/welcome/bat/bool2/CaughtSpeedingEntity.scala
package lessons.welcome.bat.bool2

import plm.universe.bat.BatEntity
import plm.universe.bat.BatTest

class CaughtSpeedingEntity extends BatEntity {

    override def run() {
      import plm.core.ValueSerializer._

      val count = getTestCount()
      for (i <- 0 to count -1) {
        val param = deserialize(getTest(i)).asInstanceOf[Array[Object]]
      	setTestResult(i, serialize(caughtSpeeding(param(0).asInstanceOf[Int], param(1).asInstanceOf[Boolean]) ))
	  }
	}

	/* BEGIN TEMPLATE */
	def caughtSpeeding(speed:Int, isBirthday:Boolean):Int = {
		/* BEGIN SOLUTION */
		if ((isBirthday && speed <= 65) || (speed <= 60))
			return 0
		else if ((isBirthday && speed <= 85) || (speed <= 80))
			return 1
		else
			return 2
		/* END SOLUTION */
	}
	/* END TEMPLATE */
}
